test/e2e/framework/framework.go

```diff
@@ -129,8 +129,17 @@ func (f *Framework) WaitForGameServerState(gs *agonesv1.GameServer, state agones
 }
 
 // WaitForFleetCondition waits for the Fleet to be in a specific condition or fails the test if the condition can't be met in 5 minutes.
-// nolint: dupl
 func (f *Framework) WaitForFleetCondition(t *testing.T, flt *agonesv1.Fleet, condition func(fleet *agonesv1.Fleet) bool) {
+	t.Helper()
+	err := f.WaitForFleetConditionParallel(t, flt, condition)
+	if err != nil {
+		// Do not call Fatalf() from go routine other than main test go routine, because it could cause a race
+		t.Fatalf("error waiting for fleet condition on fleet %v", flt.Name)
```

Comment:
Now that I'm looking at this, should this function instead return an `error` if it fails, much like we do for other `Wait*` commands ([example](https://github.com/googleforgames/agones/blob/e8692b2b0baa06a0b3c684a00017749471206e4a/test/e2e/framework/framework.go#L107)). That would make it simpler no? (and we no longer need to pass through `testing.T` either.

Reply:
There is a 73 occurrences of WaitForFleetCondition() so it is handy to enclose this `Fatalf()` function inside this helper ( if it called from main test routine).

Reply:
Then I would argue should really be an AssertFleetCondition - rather than a wait. Because it's making an assertion, rather than waiting for a result.